File: pixels-core/src/main/java/io/pixelsdb/pixels/core/exception/PixelsFileMagicInvalidException.java
package io.pixelsdb.pixels.core.exception;

/**
 * pixels
 *
 * @author guodong
 */
public class PixelsFileMagicInvalidException
        extends PixelsRuntimeException
{
    private static final long serialVersionUID = 8860468509522142835L;

    public PixelsFileMagicInvalidException(String magic)
    {
        super(String.format("The file magic %s is not valid.", magic));
    }
}

File: cpp/pixels-core/lib/exception/PixelsFileMagicInvalidException.cpp
//
// Created by liyu on 3/14/23.
//

#include "exception/PixelsFileMagicInvalidException.h"

PixelsFileMagicInvalidException::PixelsFileMagicInvalidException(const std::string &magic) {
    std::cout<<"The file magic "
        << magic
        <<" is not valid."
        <<std::endl;
}
